GAE/src/org/waterforpeople/mapping/app/web/SurveyAssemblyServlet.java

```diff
@@ -90,51 +98,50 @@ protected RestRequest convertRequest() throws Exception {
-                        .listQuestionsBySurvey(importReq.getSurveyId());
-                if (questionList != null && questionList.size() > BACKEND_QUESTION_THRESHOLD) {
-                    useBackend = true;
-                }
+            Date start = new Date();
+            Long id = importReq.getSurveyId();
+            //Instrumentation
+            ProcessingStatusDao statusDao = new ProcessingStatusDao();
+            ProcessingStatus status = statusDao.getStatusByCode(
+                    FORM_PUB_STATUS_KEY + (id != null ? ":" + id : ""));
+            if (status == null) {
+                status = new ProcessingStatus();
+                status.setCode(FORM_PUB_STATUS_KEY + (id != null ? ":" + id : ""));
+                status.setMaxDuration(0.0);
             }
-            if (useBackend) {
-                log.info("Forking to task for long assembly");
-                com.google.appengine.api.taskqueue.TaskOptions options = com.google.appengine.api.taskqueue.TaskOptions.Builder
-                        .withUrl("/app_worker/surveyassembly")
-                        .param(SurveyAssemblyRequest.ACTION_PARAM,
-                                SurveyAssemblyRequest.ASSEMBLE_SURVEY)
-                        .param(SurveyAssemblyRequest.IS_FWD_PARAM, "true")
-                        .param(SurveyAssemblyRequest.SURVEY_ID_PARAM,
-                                importReq.getSurveyId().toString());
-                // change the host so the queue invokes the backend
-                options = options
-                        .header("Host",
-                                BackendServiceFactory.getBackendService()
-                                        .getBackendAddress("dataprocessor"));
-                com.google.appengine.api.taskqueue.Queue queue = com.google.appengine.api.taskqueue.QueueFactory
-                        .getQueue("surveyAssembly");
-                queue.add(options);
-            } else {
-                assembleSurveyOnePass(importReq.getSurveyId());
+            status.setLastEventDate(start);
+            Double maxDuration = status.getMaxDuration();
+            if (maxDuration == null) {
+                maxDuration = 0.0;
             }
+            status.setInError(true); //In case it never saves an end sts
+            status.setValue("inProgress");
+            statusDao.save(status);
 
+            //Need to keep this shorter than task queue limit (600s)
+            boolean ok = assembleSurveyOnePass(importReq.getSurveyId());
+            //Clear caches
             List<Long> ids = new ArrayList<Long>();
-            ids.add(importReq.getSurveyId());
+            ids.add(id);
             SurveyUtils.notifyReportService(ids, "invalidate");
+            
+            // now update the status
+            status.setInError(ok);
+            status.setValue("finished");
+            Double duration = (new Date().getTime() - start.getTime())/1000.0;
```

Comment:
Maybe its enough to save the `duration` in milliseconds and store it as a `long`?

Reply:
Ok... guess i just liked things in seconds.